### PR TITLE
Update List and Delete sections of Developer guide

### PR DESCRIPTION
**Pull request by PureUsagi**

### CODE REVIEW

File: docs/images/DeleteSequenceDiagram.png
Comment (YHWong20) [inline docs/images/DeleteSequenceDiagram.png:1]: Might be helpful to show how `targetIndex` was derived for the `Item` with name `keyword` (perhaps using a note? Or a separate arrow and activation bar, depending on whichever is more appropriate).

**Review by YHWong20 — APPROVED** [review]
Looks good! Thanks for implementing the sequence diagrams.

**Review by nur-haziq — APPROVED** [review]
LGTM. Additions to DG is useful!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -398,6 +435,11 @@ names with the search parameter, it will store the index in the `targetIndex` va
 then call another `deleteItem(int index)` method, but this time, the parameter passed is an integer. The execution after this
 will be exactly the same as passing an `Integer` to the `deleteItem(int index)` method mentions above.
 
+Furthermore, the `deleteItem(int index)` method will be accessing a `ArrayList<Integer` called `sortedOrder` which contains
+the index mapping between the `Item` objects stored in `itemList` and the index of `Item` printed to the user. This
+functionality is implemented to ensure that the indexes of the sorted `list` shown to the user can be used as reference
```
[the remark this answers is not in the document]
Comment (PureUsagi) [reply docs/DeveloperGuide.md:440]: Yep that's a good idea, will add it in.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -275,9 +275,44 @@ access to `ItemList`.
 
 ### List all items in inventory
 
-The `ListCommand` class is designed to handle the operation of listing all items in the inventory. When the `execute()` method is called, it retrieves the complete list of items from the `ItemList`
+![ListSequenceDiagram](images/ListSequenceDiagram.png)
+
+API: [`ListCommand.java`](https://github.com/AY2324S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/binbash/command/ListCommand.java)
+
+The `ListCommand` class is designed to handle the operation of listing all items in the inventory. When the `execute()`
+method is called, it retrieves the complete list of items from the `ItemList`
```
[the remark this answers is not in the document]
Comment (PureUsagi) [reply docs/DeveloperGuide.md:283]: In this case I think it may be irrelevant as the main logic starts from the execute() method.

**Pull request merged**